x-pack/plugins/index_management/public/application/components/mappings_editor/_index.scss

```diff
@@ -15,12 +15,18 @@
 .mappingsEditor {
   &__createFieldWrapper {
     background-color: $euiColorLightestShade;
-    border-right: $euiBorderThin;
-    border-bottom: $euiBorderThin;
-    border-left: $euiBorderThin;
     padding: $euiSize;
   }
-
+  &__createFieldBorder {
+    border-left: $euiBorderThin;
+    border-right:  $euiBorderThin;
+    border-bottom:  $euiBorderThin;
+  }
+  &__createFieldAddField {
+    border-right: 0;
+    border-bottom: 0;
+    border-left: 0;
+  }
```

Comment:
I think these changes causes the top-border to disappear in the mappings editor when used for index/component templates
<img width="887" alt="Screenshot 2024-03-26 at 13 37 04" src="https://github.com/elastic/kibana/assets/6585477/00e97e38-0aff-4465-bc41-dd3df5703ec5">

Reply:
Tried to replicate same without my changes in a cloud instance, and I don't see top-border being shown in index/component template form when fields are empty. 
When a first field is added, [bottom-border is added](https://github.com/elastic/kibana/blob/main/x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/_field_list_item.scss#L13) and when add field is clicked again , the form will have borders in all four sides, where the top-border is actually bottom-border from previous field

Reply:
>  maybe we could remove this class altogether?

to confirm  -no borders for index/component template creation flow and index detail page?

Reply:
Yes, I think it would be better without borders than with only 3

Reply:
@yuliacech I suppose we would want bottom border in list view? Currently this [CSS](https://github.com/elastic/kibana/blob/main/x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/_field_list_item.scss#L13) is shared between the bottom component (listing added fields) and top component(where new fields are added) . If we remove all the borders this would remove bottom-border as well, which shows difference between each fields. 

<img width="1711" alt="Screenshot 2024-03-26 at 10 23 01 AM" src="https://github.com/elastic/kibana/assets/55930906/52f7be07-74ed-4748-97ff-5719cd142805">

Reply:
I think that is caused by the classes that you added in this PR, try to rollback any changes to the css and the added classes in the file "x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/create_field/create_field.tsx". Then try removing only right, bottom, left borders in this css file for the class "createFieldWrapper", that should result in the list still having the lines separating the fields but the form to be without borders 
<img width="1297" alt="Screenshot 2024-03-26 at 16 15 35" src="https://github.com/elastic/kibana/assets/6585477/d6327bd4-2d82-447a-8896-03d13719c1e4">
<img width="1280" alt="Screenshot 2024-03-26 at 16 15 56" src="https://github.com/elastic/kibana/assets/6585477/e5f7ae10-9a06-495d-9cd4-2076035ec259">

Reply:
<img width="455" alt="Screenshot 2024-03-26 at 16 18 32" src="https://github.com/elastic/kibana/assets/6585477/1a318136-dc7d-4011-a6c0-d9b5cdedfbe7">

Reply:
Got it. I was a bit confused that we also need to remove bottom-border as well. This makes sense. Thank you! 
Added in f883bc79cdb9bb47e14ad8319f24549e03132560